app/credentials/concerns/user_provided_services.rb

```diff
@@ -0,0 +1,17 @@
+module UserProvidedService
+  def credentials(service_name)
+    user_provided_service[service_name]
```

Comment:
It'd be hugely helpful if an exception was thrown for missing required values. AFAIK we didn't have that before, for env vars, but it would make certain setup issues much more discoverable. Perhaps we could do that with two different `credentials` methods, one for required values and one for optional values?